bool isHit
string message
int32 x
int32 y

